src/fp2.rs

```diff
@@ -422,545 +332,631 @@ impl Fp2 {
-                    0xeb208d51e08bcf27,
-                    0xe98ad40811f5fc2b,
-                    0x736c3a59232d511d,
-                    0x10acd42d29cfcbb6,
-                ]),
-                c1: Fp::from_raw_unchecked([
-                    0xd328e37cc2f58d41,
-                    0x948df0858a605869,
-                    0x6032f9d56f93a573,
-                    0x2be483ef3fffdc87,
-                    0x30ef61f88f483c2a,
-                    0x1333f55a35725be0,
-                ]),
-            }
-            .lexicographically_largest()
-        ));
-        assert!(!bool::from(
-            Fp2 {
-                c0: -Fp::from_raw_unchecked([
-                    0x1128ecad67549455,
-                    0x9e7a1cff3a4ea1a8,
-                    0xeb208d51e08bcf27,
-                    0xe98ad40811f5fc2b,
-                    0x736c3a59232d511d,
-                    0x10acd42d29cfcbb6,
-                ]),
-                c1: -Fp::from_raw_unchecked([
-                    0xd328e37cc2f58d41,
-                    0x948df0858a605869,
-                    0x6032f9d56f93a573,
-                    0x2be483ef3fffdc87,
-                    0x30ef61f88f483c2a,
-                    0x1333f55a35725be0,
-                ]),
-            }
-            .lexicographically_largest()
-        ));
-        assert!(!bool::from(
-            Fp2 {
-                c0: Fp::from_raw_unchecked([
-                    0x1128ecad67549455,
-                    0x9e7a1cff3a4ea1a8,
-                    0xeb208d51e08bcf27,
-                    0xe98ad40811f5fc2b,
-                    0x736c3a59232d511d,
-                    0x10acd42d29cfcbb6,
-                ]),
-                c1: Fp::zero(),
-            }
-            .lexicographically_largest()
-        ));
-        assert!(bool::from(
-            Fp2 {
-                c0: -Fp::from_raw_unchecked([
-                    0x1128ecad67549455,
-                    0x9e7a1cff3a4ea1a8,
-                    0xeb208d51e08bcf27,
-                    0xe98ad40811f5fc2b,
-                    0x736c3a59232d511d,
-                    0x10acd42d29cfcbb6,
-                ]),
-                c1: Fp::zero(),
+                fn visit_seq<V>(self, mut seq: V) -> Result<Fp2, V::Error>
+                where
+                    V: serde::de::SeqAccess<'de>,
+                {
+                    let c0 = seq
+                        .next_element()?
+                        .ok_or_else(|| serde::de::Error::invalid_length(0, &self))?;
+                    let c1 = seq
+                        .next_element()?
+                        .ok_or_else(|| serde::de::Error::invalid_length(0, &self))?;
+                    Ok(Fp2 { c0, c1 })
+                }
             }
-            .lexicographically_largest()
-        ));
+
+            const FIELDS: &[&str] = &["c0", "c1"];
+            deserializer.deserialize_struct("Fp2", FIELDS, Fp2Visitor)
+        }
     }
 
     #[test]
-    #[cfg(feature = "serde_req")]
```

Comment:
Does this file only compile with `serde_req`? If not this needs to stay.

Reply:
The whole `dusk` module only compiles with "serde_req" so I moved the feature flag to line 868